test/truffle/upgradedContract.ts

```diff
@@ -0,0 +1,10 @@
+import fs from 'fs';
+
+const claimManagerContract = fs.readFileSync('./contracts/roles/ClaimManager.sol', 'utf8');
+
+// Creates a different copy of the ClaimManager contract
+const upgradedClaimManagerContract = claimManagerContract
+    .replace(`contract ClaimManager is `, `contract ClaimManagerUpgradeTest is `)
+    .replace(`return "v0.1";`, `return "v0.2";`);
+
+fs.writeFileSync('./contracts/roles/ClaimManagerUpgradeTest.sol', upgradedClaimManagerContract);
```

Comment:
Could you elaborate on the purpose of this script please?

Reply:
This script creates a copy of the ClaimManager and changes the version to v2. This new contract is then deployed as an upgrade to the previous contract and validated if the upgrade is successful or not.

Reply:
Thanks, makes sense 👍